### DISCUSSION TITLE
1.26.0 Unable to log into vaultwarden after update

### DISCUSSION THREAD

**Discussion by chrispro-21** [dani-garcia/vaultwarden#2829] [Q&A] [answered]
### Subject of the issue

Unable to log in to vaultwarden after updating from 1.25.2 to 1.26.0. I went back to 1.25.2 as logging in still works, but I got a `Username or Password is incorrect. Try again. IP: <my IP address>. Username: <email>.`
### Deployment environment

* vaultwarden version: 1.26.0

* Install method: docker

* Clients used: web vault 

* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details: single account on server, caddy reverse proxy outside of docker network.

### Steps to reproduce

 Update from 1.25.2 to 1.26.0 on docker 
### Expected behavior

a successful login of an account
### Actual behavior

Unable to login with error: `Username or Password is incorrect. Try again. IP: <my IP address>. Username: <email>.`
### Troubleshooting data

```
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                           Version 1.26.0                           |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[INFO] No .env file found.

[2022-10-14 21:49:22.983][start][INFO] Rocket has launched from http://0.0.0.0:80
[2022-10-14 21:49:30.811][request][INFO] POST /api/accounts/prelogin
[2022-10-14 21:49:30.812][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
[2022-10-14 21:49:30.834][request][INFO] POST /identity/connect/token
[2022-10-14 21:49:30.834][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: x.x.x.x. Username: xxxxxxx@xxxxxx.com.
[2022-10-14 21:49:30.834][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2022-10-14 21:51:40.838][request][INFO] POST /api/accounts/prelogin
[2022-10-14 21:51:40.840][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK


```

**Comment by BlackDex** [comment]
Which database are you using?
What do you see with both versions in the admin interface at the users section?

Also, could you provide the support string of at least the working version, and if possible also the non-working.

**Comment by kb8cb40j1av39c02** [comment]
Experienced the same issue with a similar setup. Standard Vaultwarden installation with SQLite. My container got auto-updated by watchtower as do other containers.
I was able to fix the issue by rebooting my docker host. I also recall having to do that last time after Vaultwarden got updated.
Strange behaviour that isn't expexted.

**Comment by zaidi-01** [comment]
I'm having this issue as well on 1.26.0. The `/identity/connect/token` endpoint keeps throwing a 400 and I can't access my vault at all.

```
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                           Version 1.26.0                           |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/
[INFO] No .env file found.
[DEPRECATED]: `SMTP_SSL` or `SMTP_EXPLICIT_TLS` is set. Please use `SMTP_SECURITY` instead.
[2022-10-15 05:55:53.688][start][INFO] Rocket has launched from http://0.0.0.0:80
[2022-10-15 05:57:20.939][request][INFO] POST /api/accounts/prelogin
[2022-10-15 05:57:20.948][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
[2022-10-15 05:57:21.050][request][INFO] POST /identity/connect/token
[2022-10-15 05:57:21.153][error][ERROR] 2FA token not provided
[2022-10-15 05:57:21.153][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2022-10-15 05:59:44.200][request][INFO] POST /api/accounts/prelogin
[2022-10-15 05:59:44.201][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
[2022-10-15 05:59:44.272][request][INFO] POST /identity/connect/token
[2022-10-15 05:59:44.362][error][ERROR] 2FA token not provided
[2022-10-15 05:59:44.362][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
```

**Comment by BlackDex** [comment]
> I'm having this issue as well on 1.26.0. The `/identity/connect/token` endpoint keeps throwing a 400 and I can't access my vault at all.
> 
> ```
> /--------------------------------------------------------------------\
> |                        Starting Vaultwarden                        |
> |                           Version 1.26.0                           |
> |--------------------------------------------------------------------|
> | This is an *unofficial* Bitwarden implementation, DO NOT use the   |
> | official channels to report bugs/features, regardless of client.   |
> | Send usage/configuration questions or feature requests to:         |
> |   https://vaultwarden.discourse.group/                             |
> | Report suspected bugs/issues in the software itself at:            |
> |   https://github.com/dani-garcia/vaultwarden/issues/new            |
> \--------------------------------------------------------------------/
> [INFO] No .env file found.
> [DEPRECATED]: `SMTP_SSL` or `SMTP_EXPLICIT_TLS` is set. Please use `SMTP_SECURITY` instead.
> [2022-10-15 05:55:53.688][start][INFO] Rocket has launched from http://0.0.0.0:80
> [2022-10-15 05:57:20.939][request][INFO] POST /api/accounts/prelogin
> [2022-10-15 05:57:20.948][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
> [2022-10-15 05:57:21.050][request][INFO] POST /identity/connect/token
> [2022-10-15 05:57:21.153][error][ERROR] 2FA token not provided
> [2022-10-15 05:57:21.153][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
> [2022-10-15 05:59:44.200][request][INFO] POST /api/accounts/prelogin
> [2022-10-15 05:59:44.201][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
> [2022-10-15 05:59:44.272][request][INFO] POST /identity/connect/token
> [2022-10-15 05:59:44.362][error][ERROR] 2FA token not provided
> [2022-10-15 05:59:44.362][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
> ```

That isn't an error actually, it indicates you need to provide a 2fa/mfa token before you are allowed to continue to login.

**Comment by BlackDex** [comment]
> Experienced the same issue with a similar setup. Standard Vaultwarden installation with SQLite. My container got auto-updated by watchtower as do other containers. I was able to fix the issue by rebooting my docker host. I also recall having to do that last time after Vaultwarden got updated. Strange behaviour that isn't expexted.

Needing to reboot your host is not something which can be solved by a container. I have the same kind of setup with watchtower and using the testing tagged images. I haven't experienced any issues.
In your case i would suggest to check you dockerd/host logs to see what went wrong. Could be firewalling, kernel libraries which were updated, other applications which were updated,

I really doubt it is the fault of a container. But only logs could help to try and determine it.

**Comment by kb8cb40j1av39c02** [comment]
@BlackDex from my mobile phone I was only able to just quickly restart the host and not the container itself. I'm not sure if just rebooting the container would have fixed the issue too.

**Comment by Bloodpack** [comment]
Same Problem here after update to 1.26.0

reboot container and host system does NOT fix the problem here

```bash
/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                           Version 1.26.0                           |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/
[INFO] No .env file found.
[2022-10-15 13:37:58.954][start][INFO] Rocket has launched from http://0.0.0.0:80
[2022-10-15 13:38:17.275][request][INFO] POST /api/accounts/prelogin
[2022-10-15 13:38:17.275][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
[2022-10-15 13:38:17.349][request][INFO] POST /identity/connect/token
[2022-10-15 13:38:17.350][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: XXX.XXXX.XXX.X. Username: myemail@email.provider.
[2022-10-15 13:38:17.350][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2022-10-15 13:39:59.582][request][INFO] POST /api/accounts/prelogin
[2022-10-15 13:39:59.583][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
[2022-10-15 13:39:59.652][request][INFO] POST /identity/connect/token
[2022-10-15 13:39:59.653][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: XXX.XXXX.XXX.X. Username: myemail@email.provider.
[2022-10-15 13:39:59.653][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2022-10-15 13:40:25.908][request][INFO] POST /api/accounts/prelogin
[2022-10-15 13:40:25.908][response][INFO] (prelogin) POST /api/accounts/prelogin => 200 OK
[2022-10-15 13:40:25.986][request][INFO] POST /identity/connect/token
[2022-10-15 13:40:25.987][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: XXX.XXXX.XXX.X. Username: myemail@email.provider.
[2022-10-15 13:40:25.987][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2022-10-15 13:41:41.202][request][INFO] POST /identity/connect/token
[2022-10-15 13:41:41.211][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2022-10-15 13:42:10.237][request][INFO] POST /identity/connect/token
[2022-10-15 13:42:10.237][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
```

**Comment by BlackDex** [comment]
Which web-vault version are you using?
Which reverse proxy, some reverse proxies block some stuff or could be wrongly configured.
All i can say is that it works just fine for a long time now, since the code-base of 1.26.0 has been out for a while already under testing.

Also, it says login incorrect, so please double check via the `/admin` interface if all the correct users are there or not.
Also, post the `Support String` which you can generate via the admin interface under the `Diagnostics` page.

**Comment by Bloodpack** [comment]
@BlackDex 

thank you for the reply

i solved my problem.....

i deleted the vaultwarden container and installed it again and now it works (don´t ask me why, i don´t know)
updating from 1.25.2 did break it....

i am using pfsense HA-Proxy, but i used it for a long time with no problems

**Comment by chrispro-21** [comment]
I'll try a reinstall next and report back.

**Comment by chrispro-21** [comment, marked as answer]
The reinstall appeared to work after exporting my vaults to json and deleting the /vw-data/ and recreating it.

Thank you!

**Reply by Xanderlicious** [reply]
> The reinstall appeared to work after exporting my vaults to json and deleting the /vw-data/ and recreating it.
> 
> Thank you!

I am having this problem and have recreated my docker container already - can you explain what you did with regards to exorting your vault to json etc

**Comment by BlackDex** [comment]
Strange though.
Almost looks like it didn't used the old database and created a new one without overwriting the old.

Glad you still have all the data and it works now.

**Comment by Bloodpack** [comment]
@chrispro-21 
I did not have to delete my persistent data folders and it worked

**Reply by chrispro-21** [reply]
Interesting. I do have snapshots + full backups of the VM if I need to go back or spin up another for testing.

**Comment by Xanderlicious** [comment]
Well that was a horrific couple of hours trying to sort that out.

If it helps anyone.....

I found that when installing Bitwarden again it created new data and placed my existing data into a /data folder

as soon as I pointed my volume to here, I was able to log in fine

**Comment by makhlouta** [comment]
I had the same issue, portainer was showing vaultwarden-old as exited and vaultwarden as created but it wouldn't start. Rebooting the host or container didn't help. I deleted and recreated it (without deleting persistent data) and things are back to normal

**Comment by mahyess** [comment]
I just found this problem with my Caprover hosted Vaultwarden. (I didn't have to login in a new extension until now.)

I deleted the app, but without the persistent data. Created another app with the same name, and it just worked.

**Comment by diagpope** [comment]
I discovered this problem yesterday as well:
1. No vaultwarden or bitwarden docker allows login from any extensions: Safari, Chrome/Brave, Firefox anymore
2. Pruning images system, removing container and reinstaling makes no difference, also clearing vaults and starting from scratch has no effect 
Web login works but not from any browser extension. What is the secret remedy?

**Reply by stefan0xC** [reply]
See https://github.com/dani-garcia/vaultwarden/issues/3082

**Reply by diagpope** [reply]
Thanks, I reinstalled via https://www.vultr.com/docs/how-to-install-vaultwarden-on-ubuntu-20-04/ and all is well again.
